cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+            st.executeUpdate(alterDestTableDDL);
+        }
+        else {
+            LOGGER.debug("No additional columns have been found");
+        }
+    }
+
+    private void execDestTableDDL(String destDb,String destHiveConnection) throws Exception{
+        Connection conn = getHiveJDBCConnection(destDb,destHiveConnection);
+        Statement st = conn.createStatement();
+        LOGGER.debug("Destination table not found. Hence creating one");
+        st.executeUpdate(destTableDDL);
+        st.close();
+        conn.close();
+    }
+
+    private String formFilterCondition(Set<String> modifiedBusinessPartitionSet, String sourcePartitionColumns){
+        StringBuffer filterCondition=new StringBuffer();
+        Map<String,String> partitionDataTypeMap = new HashMap<>();
+        String[] partitionArray = sourcePartitionColumns.split(",");
+        for(int i=0;i<partitionArray.length;i++){
+            partitionDataTypeMap.put(partitionArray[i].toUpperCase().substring(0,partitionArray[i].indexOf(" ")),partitionArray[i].toUpperCase().substring(partitionArray[i].indexOf(" "),partitionArray[i].length()));
+        }
+        for(String busPartition:modifiedBusinessPartitionSet){
+            String onePartition="";
+            String[] eachPartitionValue = busPartition.split("/");
+            StringBuffer totalRow=new StringBuffer("");
+            for(int i=0; i<eachPartitionValue.length;i++){
+                String partitionDataType=partitionDataTypeMap.get(eachPartitionValue[i].split("=")[0].toUpperCase());
+                String colAndPartitionValue[]=eachPartitionValue[i].split("=");
+                if(partitionDataType.trim().contains("STRING")||partitionDataType.trim().contains("CHAR")||partitionDataType.trim().contains("DATE")||partitionDataType.trim().contains("TIME")) {
+                    colAndPartitionValue[1]="'"+colAndPartitionValue[1]+"'";
+                }
+                onePartition=colAndPartitionValue[0]+"="+colAndPartitionValue[1] +" AND ";
+                totalRow.append(onePartition);
+            }
+            String trimmedTotalRow=totalRow.substring(0,totalRow.lastIndexOf(" AND "));
+            filterCondition.append(trimmedTotalRow).append(" OR ");
+        }
+
+        return(filterCondition.toString().isEmpty()?"'a'='a'":filterCondition.substring(0,filterCondition.lastIndexOf(" OR ")));
+    }
+
+    private String removeDataTypesFromColumnList(String commaSeparatedColumns){
+        StringBuffer columnListWithoutDataTypes= new StringBuffer("");
+        String[] columns = commaSeparatedColumns.split(",");
+        for(int i=0;i<columns.length;i++){
+            columnListWithoutDataTypes.append(columns[i].substring(0,columns[i].indexOf(" "))).append(",");
+        }
+        return columnListWithoutDataTypes.toString().substring(0,columnListWithoutDataTypes.length()-1);
+    }
+
+    private String getTableLocation(String sourceDb,String hiveConnection,String sourceStgTable) throws Exception{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)